### PR TITLE
allow filtering conversations list by type

### PR DESCRIPTION
**Pull request by davemeyer**
support the filters slack provides including
* conversation type
* shared channels
* bot users

in the `<ConversationsSelect />` component

### CODE REVIEW

File: src/components/ConversationsSelect.ts [hunk 1]
```diff
@@ -6,6 +6,13 @@ import { FC } from '..';
 export interface ConversationSelectProps extends ContainerProps<string> {
   initialConversation?: string;
   actionId: string;
+  filter?: {
+    include?: ('im' | 'mpim' | 'private' | 'public')[];
```
Comment (mshende) [inline src/components/ConversationsSelect.ts:10]: what's the difference between these? obviously 'public' is public, but the other three?

Comment (davemeyer) [reply src/components/ConversationsSelect.ts:10]: public -> public channels
private -> private channels
im -> direct message with just two people
mpim -> multi-person direct message, a group chat

the block fields can be seen here https://api.slack.com/reference/block-kit/composition-objects#filter_conversations

File: src/components/ConversationsSelect.ts [hunk 2]
```diff
@@ -16,13 +23,21 @@ export const ConversationsSelect: FC<
...
 }) => {
   const select: ConversationsSelectSpec = {
     type: 'conversations_select',
     initial_conversation: initialConversation,
     action_id: actionId,
     response_url_enabled: responseUrlEnabled,
+    ...(filter && {
+      filter: {
+        include: filter.include,
+        exclude_bot_users: filter.excludeBotUsers,
```
Comment (andycmaj) [inline src/components/ConversationsSelect.ts:37]: having these nested filter fields be undefined is not an issue for slack API?

**Review by andycmaj — APPROVED** [review]

File: src/components/ConversationsSelect.ts (hunk 2 above)
Comment (davemeyer) [reply src/components/ConversationsSelect.ts:37]: it didn't cause any issues in my testing. they appear to treat it as false

**Pull request merged**